### ISSUE TITLE
Regression: Cors::beforeAction()

### ISSUE THREAD

**Issue by alexraputa** [yiisoft/yii2#15665]
There's a regression in 2.0.14-dev `Cors::beforeAction()` after merge: https://github.com/yiisoft/yii2/commit/399dbce0cadbd7631f726841ece3ecf6a830444d

This is my test in Codeception:
```php
public function preflight(ApiTester $I): void
{
    $I->haveHttpHeader('Access-Control-Request-Headers', 'Content-Type');
    $I->haveHttpHeader('Access-Control-Request-Method', 'POST');
    $I->haveHttpHeader('Accept', '*/*');
    $I->sendOPTIONS('/users');

    $I->seeResponseCodeIs(200);
    $I->seeHttpHeader('Content-Type', 'application/vnd.api+json; charset=UTF-8');
    $I->seeHttpHeader('Access-Control-Allow-Methods', 'GET, POST, PUT, PATCH, DELETE, HEAD, OPTIONS');
    $I->seeHttpHeader('Access-Control-Allow-Headers', 'Content-Type');
}
```

Before merge https://github.com/yiisoft/yii2/commit/399dbce0cadbd7631f726841ece3ecf6a830444d:

```
See http header "Content-Type","application/vnd.api+json; charset=UTF-8"

- Expected | + Actual
@@ @@
-'application/vnd.api+json; charset=UTF-8'
+'application/vnd.api+json; charset=UTF-8'
```

After merge https://github.com/yiisoft/yii2/commit/399dbce0cadbd7631f726841ece3ecf6a830444d:

```
See http header "Content-Type","application/vnd.api+json; charset=UTF-8"

- Expected | + Actual
@@ @@
-'application/vnd.api+json; charset=UTF-8'
+'text/html; charset=UTF-8'
```

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.14-dev
| PHP version      | 7.1.4
| Operating system | macOS High Sierra 10.13.3

**Comment by samdark** [comment]
What's the actual code that is tested?

**Comment by alexraputa** [comment]
@samdark, this is an action that I testing:
```php
public function actionCreate()
{
    $resource = new SignUpResource();

    if ($resource->load($this->request->post()) && $resource->validate()) {
        try {
            $this->service->create($resource);
            $this->response->setStatusCode(201);
            $this->response->getHeaders()->set('Location', Url::to('users/' . $resource->id, true));

            return $resource;
        } catch (\DomainException $e) {
            \Yii::$app->errorHandler->logException($e);
        }
    } elseif ($resource->hasErrors()) {
        return $resource;
    }
}
```

**Comment by alexraputa** [comment]
rest controller:
```php
class BaseController extends \yii\web\Controller
{
...
    public $serializer = \tuyakhov\jsonapi\Serializer::class;

    public $enableCsrfValidation = false;

    public function behaviors(): array
    {
        return [
            'contentNegotiator' => [
                'class' => \yii\filters\ContentNegotiator::class,
                'formats' => [
                    'application/vnd.api+json' => \yii\web\Response::FORMAT_JSON,
                    'application/json' => \yii\web\Response::FORMAT_JSON,
                    'application/xml' => \yii\web\Response::FORMAT_XML,
                ],
            ],
            'verbFilter' => [
                'class' => \yii\filters\VerbFilter::class,
                'actions' => $this->verbs(),
            ],
            'rateLimiter' => [
                'class' => \yii\filters\RateLimiter::class,
            ],
        ];
    }
```

config:

```php
'components' => [
    'request' => [
        'parsers' => [
            'application/vnd.api+json' => [
                'class' => tuyakhov\jsonapi\JsonApiParser::class,
                'memberNameCallback' => [tuyakhov\jsonapi\Inflector::class, 'variablize'],
            ],
            'application/json' => yii\web\JsonParser::class,
        ],
    ],
    'response' => [
        'formatters' => [
            yii\web\Response::FORMAT_JSON => [
                'class' => tuyakhov\jsonapi\JsonApiResponseFormatter::class,
                'prettyPrint' => YII_DEBUG,
                'encodeOptions' => JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE,
            ],
        ],
    ],
    'urlManager' => [
        'enablePrettyUrl' => true,
        'enableStrictParsing' => true,
        'showScriptName' => false,
        'rules' => [
            ...
            'OPTIONS,GET users' => 'users/index',
            'OPTIONS,POST users' => 'users/create',
        ],
    ],
],
'as corsFilter' => [
    'class' => yii\filters\Cors::class,
],
```

**Comment by samdark** [comment]
It's not clear how to reproduce it. There are custom classes everywhere and excessive config. Would you please do a minimal test case that contains only necessary code to reproduce the issue starting with basic app?

**Comment by alexraputa** [comment]
@samdark, I'll try.

**Comment by alexraputa** [comment]
@samdark, the issue is reproduce in the basic app [cors-bug.zip](https://github.com/yiisoft/yii2/files/1735823/cors-bug.zip).

How to reproduce the issue.
1. `composer install`
2. `php vendor/bin/codecept run api`
The result should be fail.
3. These lines should be comment out. https://github.com/yiisoft/yii2/blob/4f80cda7130da5259ce93fe41eb681cbb70f30be/framework/filters/Cors.php#L109-L113
The result should be success.

**Comment by leandrogehlen** [comment]
This is strange bacause Cors filter is not enabled in this project

**Comment by samdark** [comment]
@leandrogehlen and you confirm that commenting lines above make tests pass?

**Comment by leandrogehlen** [comment]
I'm looking for it now

**Comment by alexraputa** [comment]
w/ https://github.com/yiisoft/yii2/commit/399dbce0cadbd7631f726841ece3ecf6a830444d:
<img width="781" alt="2018-02-19 19 43 27" src="https://user-images.githubusercontent.com/10957915/36391067-989055a2-15ad-11e8-856d-956c2c5cb847.png">

w/o https://github.com/yiisoft/yii2/commit/399dbce0cadbd7631f726841ece3ecf6a830444d:
<img width="716" alt="2018-02-19 19 44 34" src="https://user-images.githubusercontent.com/10957915/36391068-98ac7b92-15ad-11e8-89ad-0899856db602.png">

**Comment by leandrogehlen** [comment]
I did some tests and the unique issue is that if you comment  second line the test works.
Then can't see the problem, because the header: `'Content-Type'` is not set in request and `ContentNegotiator` not add this header it in response. I think this is right.

```php
   $I->seeResponseCodeIs(200);
   //$I->seeHttpHeader('Content-Type', 'application/json; charset=UTF-8');
   $I->seeHttpHeader('Access-Control-Allow-Methods', 'GET, POST, PUT, PATCH, DELETE, HEAD, OPTIONS');
   $I->seeHttpHeader('Access-Control-Allow-Headers', 'Content-Type');

**Comment by leandrogehlen** [comment]
I think i found the problem, in example project was configured in `web.php`:
```php  
   'as corsFilter' => [
        'class' => yii\filters\Cors::class,
    ],
   'as authenticator' => [
        'class' => yii\filters\auth\CompositeAuth::class,
        'except' => ['site/index', 'debug/*'],
        'authMethods' => [
            ['class' => yii\filters\auth\HttpBearerAuth::class],
        ],
    ],
```
This way behavior method is not considered. I removed this configuration and did it in controller :

```php
class SiteController extends Controller
{
    /**
     * @inheritdoc
     */
    public function behaviors()
    {
        return [
            'contentNegotiator' => [
                'class' => ContentNegotiator::class,
                'formats' => [
                    'application/json' => Response::FORMAT_JSON,
                ],
            ],
            'cors' => [
                'class' => Cors::class,
            ]
        ]
   ...
```
I change the test to 
```php
    $I->sendOPTIONS('/site/index?_format=json');
```
This worked, but i think the problem is the sequence, like `cors` return false the following filters are not executed

**Comment by samdark** [comment]
Well, that's expected and exactly that was changed, isn't it?

**Comment by alexraputa** [comment]
> This worked, but i think the problem is the sequence, like cors return false the following filters are not executed

This is the correct statement.

**Comment by leandrogehlen** [comment]
@samdark, i think so.
@alexraputa if you change the filters sequence your problem is  solved?

**Comment by leandrogehlen** [comment]
@alexraputa why do you need these headers or authentication in preflight?

**Comment by alexraputa** [comment]
@leandrogehlen, this is definitely a bug.
The preflight response should contain `Content-Type`, which will be the same as for a real response.
See [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) the section of `Preflighted requests` and example for it.

**Comment by leandrogehlen** [comment]
I agreee, but this is happening. You don't adds the 'Content-Type' in you request test.

If you use the code below, will work

```php
$I->haveHttpHeader('Content-Type', 'application/json; charset=UTF-8');
$I->haveHttpHeader('Access-Control-Request-Headers', 'Content-Type');
$I->haveHttpHeader('Access-Control-Request-Method', 'POST');
$I->haveHttpHeader('Accept', '*/*');
$I->sendOPTIONS('/users');
```

**Comment by alexraputa** [comment]
@leandrogehlen, I disagree with it. The `Content-Type` of the request specifies the request body, which is not applicable to an OPTIONS request.
See [OPTIONS](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS) request.

I use behaviors in this sequence. See [Yii 2 CORS](http://www.yiiframework.com/doc-2.0/guide-rest-controllers.html#cors).
<img width="220" alt="2018-02-20 19 38 05" src="https://user-images.githubusercontent.com/10957915/36439445-a0f10fd4-1675-11e8-8ae9-c58ab6f4a403.png">

The behaviors will not work if the first of them return `false` in the `beforeAction` method.

@samdark, I think we should revert this https://github.com/yiisoft/yii2/commit/399dbce0cadbd7631f726841ece3ecf6a830444d feature, until we find a better solution.

**Comment by leandrogehlen** [comment]
@alexraputa, I understand but i belive the interrupt in filter sequence is another issue, not associate to Cors. If it will considered a bug is a problem for all filters

**Comment by leandrogehlen** [comment]
@samdark, I can't see problem, does not make sense execute others filters in preflight, because after preflight the real request will be executed

**Comment by alexraputa** [comment]
@leandrogehlen, @samdark, Such frameworks as Laravel and Symfony does not stop the application in the cors filter if the request has OPTIONS method. They simply modify the response object and pass it further along the chain of middlewares. Why not do the same in Yii? We should not leave this as is.

**Comment by leandrogehlen** [comment]
@alexraputa, isn't to OPTIONS method is only a preflight:
https://github.com/yiisoft/yii2/blob/master/framework/filters/Cors.php#L109

**Comment by leandrogehlen** [comment]
related #14618

**Comment by alexraputa** [comment]
@leandrogehlen, The preflight request is equal to OPTIONS request.

> Unlike “simple requests”, "preflighted" requests first send an HTTP request by the OPTIONS method to the resource on the other domain, in order to determine whether the actual request is safe to send. Cross-site requests are preflighted like this since they may have implications to user data.

Example of a request from documentation:
```
OPTIONS /resources/post-here/ HTTP/1.1
Host: bar.other
User-Agent: Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10.5; en-US; rv:1.9.1b3pre) Gecko/20081130 Minefield/3.1b3pre
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Language: en-us,en;q=0.5
Accept-Encoding: gzip,deflate
Accept-Charset: ISO-8859-1,utf-8;q=0.7,*;q=0.7
Connection: keep-alive
Origin: http://foo.example
Access-Control-Request-Method: POST
Access-Control-Request-Headers: X-PINGOTHER, Content-Type
```

Example of a response from documentation:
```
HTTP/1.1 200 OK
Date: Mon, 01 Dec 2008 01:15:39 GMT
Server: Apache/2.0.61 (Unix)
Access-Control-Allow-Origin: http://foo.example
Access-Control-Allow-Methods: POST, GET, OPTIONS
Access-Control-Allow-Headers: X-PINGOTHER, Content-Type
Access-Control-Max-Age: 86400
Vary: Accept-Encoding, Origin
Content-Encoding: gzip
Content-Length: 0
Keep-Alive: timeout=2, max=100
Connection: Keep-Alive
Content-Type: text/plain
```

Example of a real request from documentation:
```
POST /resources/post-here/ HTTP/1.1
Host: bar.other
User-Agent: Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10.5; en-US; rv:1.9.1b3pre) Gecko/20081130 Minefield/3.1b3pre
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Language: en-us,en;q=0.5
Accept-Encoding: gzip,deflate
Accept-Charset: ISO-8859-1,utf-8;q=0.7,*;q=0.7
Connection: keep-alive
X-PINGOTHER: pingpong
Content-Type: text/xml; charset=UTF-8
Referer: http://foo.example/examples/preflightInvocation.html
Content-Length: 55
Origin: http://foo.example
Pragma: no-cache
Cache-Control: no-cache

<?xml version="1.0"?><person><name>Arun</name></person>
```

Example of a real response from documentation:
```
HTTP/1.1 200 OK
Date: Mon, 01 Dec 2008 01:15:40 GMT
Server: Apache/2.0.61 (Unix)
Access-Control-Allow-Origin: http://foo.example
Vary: Accept-Encoding, Origin
Content-Encoding: gzip
Content-Length: 235
Keep-Alive: timeout=2, max=99
Connection: Keep-Alive
Content-Type: text/plain

[Some GZIP'd payload]
```

Both responses have the same content type - **`Content-Type: text/plain`**.

@samdark, need your thoughts.

**Comment by alexraputa** [comment]
This https://github.com/yiisoft/yii2/commit/399dbce0cadbd7631f726841ece3ecf6a830444d feature also breaks BC if you have actionOptions like this:

```php
public function actionOptions(): void
{
    if (!$this->request->getIsOptions()) {
        $this->response->setStatusCode(405);
    }

    $options = implode(', ', array_reduce($this->verbs(), 'array_merge', []));
    $this->response->getHeaders()->set('Allow', $options);
    $this->response->getHeaders()->set('Access-Control-Allow-Method', $options);
}
```

Because the response was stopped in the `beforeAction()` method of the CORS filter. And your response will not contain the Allow and the modifiedAccess-Control-Allow-Method headers for specific actions of controllers.

**Comment by samdark** [comment]
@alexraputa according to CORS spec https://www.w3.org/TR/cors/#preflight-request the only considered headers are CORS-related. The rest of the headers and body are simply ignored so it doesn't make any sense to send them.

**Comment by samdark** [comment]
Why would you need to send `Allow` or `Content-type` if they're simply ignored?

**Comment by alexraputa** [comment]
> Why would you need to send Allow or Content-type if they're simply ignored?

@samdark, This is not true. The response from the server must support these headers. See [rfc7231](https://tools.ietf.org/html/rfc7231#page-31) and [OPTIONS](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS)
W/ this https://github.com/yiisoft/yii2/commit/399dbce0cadbd7631f726841ece3ecf6a830444d feature.
My application on Ember.js 3.0 sends a preflight request:

<img width="427" alt="request" src="https://user-images.githubusercontent.com/10957915/36524375-750c473c-17ae-11e8-8c7e-0a9e6fd1d34c.png">

And backend app on Yii 2 sends response:

<img width="426" alt="response" src="https://user-images.githubusercontent.com/10957915/36524440-ac84403e-17ae-11e8-8c53-ad0d8002d95d.png">

W/o this https://github.com/yiisoft/yii2/commit/399dbce0cadbd7631f726841ece3ecf6a830444d feature.

<img width="427" alt="response2" src="https://user-images.githubusercontent.com/10957915/36525988-07f2912c-17b5-11e8-8936-5bf5991cd29c.png">

This is a standard preflight request to the server. The other frontend frameworks such as React and Angular send the same  preflight request.
If frontend app send the `Accept: */*` (see [Accept](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept)) header then backend app must send the allowed `Content-Type` for a URL.

**Comment by samdark** [comment]
Well, this is not how preflight should work according to its specification: https://www.w3.org/TR/cors/#preflight-request

**Comment by alexraputa** [comment]
My API is public and clients are using it. If the `Accept` header receives the response with `Content-Type` like`text/html`, then clients accordingly need to implement a serializer of this type of content. But then the API sends a real response with the type `application/vnd.api+json` and the frontend application throws an error, because the serializer for `application/vnd.api+json` is not implemented.

**Comment by leandrogehlen** [comment]
@alexraputa, this is not true, i have built a angular application with yii2 cors filters and worked fine.

Preflight results:
![preflight](https://user-images.githubusercontent.com/1750751/36561587-e08bdc9a-17f2-11e8-8fa5-76c433ad8c4a.PNG)

Real request:
![real-request](https://user-images.githubusercontent.com/1750751/36561596-ebab3f3a-17f2-11e8-8d6e-510feee3f78e.PNG)

Yii2 Angular Project
[yii2-angular.zip](https://github.com/yiisoft/yii2/files/1749396/yii2-angular.zip)

```shell
cd angular
npm install
ng build --watch
```

to access angular app use: http://localhost/yii2-angular/web/dist/

**Comment by alexraputa** [comment]
@leandrogehlen, I don't see any of CORS methods in your response.
The preflight request shouldn't have the `Content-Type` header because it has an empty body.
But you force added the `Content-Type` header into the request. 

```javascript
let headers = new HttpHeaders({
  'Content-Type': 'application/json'
});
```

**Comment by leandrogehlen** [comment]
@samdark, please close this issue, he wants to have a response default format.
it has nothing to do with issue.

**Comment by alexraputa** [comment]
Ok, if you want have some rare bugs in a framework then just leave it.

**Comment by samdark** [comment]
@alexraputa which client do you have that doesn't follow CORS specification and looks at non-CORS headers in preflight?

**Comment by alexraputa** [comment]
@samdark, Which the headers of the my request do not match with the CORS specification?

- `Accept`;
- `Referer`;
- `Access-Control-Request-Headers`;
- `Origin`;
- `User-Agent`;
- `Access-Control-Request-Method`.

**Comment by samdark** [comment]
Not the CORS specification itself but [preflight part of it](https://www.w3.org/TR/cors/#preflight-request) will just ignore what's not needed for CORS:

- Exclude the author request headers.
- Exclude user credentials.
- Exclude the request entity body.

What matters (and is cached in browsers) in CORS reponse is:

- origin
- url
- max-age
- credentials
- method

That's it. Nothing about content-type or anything else. Body doesn't matter as well.

So I'm asking again, what's the client that you're using that needs `Accept` header in the response for preflight request?

**Comment by alexraputa** [comment]
@samdark,

> My application on **`Ember.js 3.0`** sends a preflight request:

These are preflight request specification in `ASP.NET` ([Preflight-requests](https://docs.microsoft.com/en-us/aspnet/core/security/cors#preflight-requests)):

> - The application doesn't set any request headers other than Accept, Accept-Language, Content-Language, Content-Type, or Last-Event-ID, and
> 
> - The Content-Type header (if set) is one of the following:
> application/x-www-form-urlencoded
> multipart/form-data
> text/plain

You can close this issue If you think that this is not a bug.

**Comment by samdark** [comment]
You're confusing request and response. See response example there:

```
HTTP/1.1 200 OK
Cache-Control: no-cache
Pragma: no-cache
Content-Length: 0
Access-Control-Allow-Origin: http://myclient.azurewebsites.net
Access-Control-Allow-Headers: x-my-custom-header
Access-Control-Allow-Methods: PUT
Date: Wed, 20 May 2015 06:33:22 GMT
```

According to W3C spec, `Cache-Control: no-cache`, `Pragma: no-cache` and `Date` are ignored and aren't cached on clients so it comes to:

```
HTTP/1.1 200 OK
Content-Length: 0
Access-Control-Allow-Origin: http://myclient.azurewebsites.net
Access-Control-Allow-Headers: x-my-custom-header
Access-Control-Allow-Methods: PUT
```

What exactly happened to Ember.js 3.0 app when we started to send these reduced CORS responses? Did it break somehow? How exactly?

**Comment by tunecino** [comment]
@alexraputa according to that

**Comment by tunecino** [comment]
> These are preflight requests specification in ASP.NET (Preflight-requests)

@alexraputa  according to that linked documentation those are not specifications but conditions that when meet the browser will omit or skip the process of sending a preflight request.

**Comment by samdark** [comment]
Those are conditions similar to what's on W3C except examples are a bit weird. They're OK but, as I said, they define request, not response.

**Comment by alexraputa** [comment]
@samdark, what do you think about it [Preflighted requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) (see example of preflight request/response)?
And about this https://github.com/yiisoft/yii2/issues/15665#issuecomment-367489876.
We already have the `actionOptions` method in the `ActiveController` class. The headers of this action are never included in the response if you use the CORS filter.

https://github.com/yiisoft/yii2/blob/77ad6bc00847d4964a0b2a82d3b70dcd7cb5a1cf/framework/rest/ActiveController.php#L102-L104
https://github.com/yiisoft/yii2/blob/77ad6bc00847d4964a0b2a82d3b70dcd7cb5a1cf/framework/rest/OptionsAction.php#L32-L44

> What exactly happened to Ember.js 3.0 app when we started to send these reduced CORS responses? Did it break somehow? How exactly?

The front-end application is working. But the response from the API was significantly reduced, which does not give a full submission of the resource.

**Comment by samdark** [comment]
I think that these response examples are OK as well but, as pointed out by specification (W3C one), non-CORS headers are not taken into account. Means you can send whatever you want: any headers, body... anything. As long as response code is 2xx and CORS headers are there, it's fine.

**Comment by samdark** [comment]
About https://github.com/yiisoft/yii2/issues/15665#issuecomment-367489876. Regular OPTIONS request and preflight request are two different things. There's no need to give out usual options response in CORS preflight response.

**Comment by samdark** [comment]
> The front-end application is working. But the response from the API was significantly reduced, which does not give a full submission of the resource.

Would you please elaborate "does not give a full submission of the resource"?

**Comment by alexraputa** [comment]
@samdark, https://github.com/yiisoft/yii2/issues/15665#issuecomment-367602078.

> You can close this issue If you think that this is not a bug.

**Comment by samdark** [comment]
I don't get it :( Does this change affect non-CORS responses?

**Comment by alexraputa** [comment]
No, it's no affect. But it's confusing.

I don't modify any response on API, but framework send the `Content-Type`: `text/html`.

<img width="883" alt="preflight-requests" src="https://user-images.githubusercontent.com/10957915/36594050-37f28b2a-18a5-11e8-8fe0-1c72ef8e4501.png">
<img width="535" alt="preflight-request" src="https://user-images.githubusercontent.com/10957915/36594049-37d3782a-18a5-11e8-9dd4-c04a4549c639.png">

**Comment by samdark** [comment]
I see. So there's no error per se and it's overall just confusing, right?

**Comment by samdark** [comment]
If so, moving to 2.0.15 to think if it can be done better.

**Comment by alexraputa** [comment]
> I see. So there's no error per se and it's overall just confusing, right?

Yes, it's right.

**Comment by alnidok** [comment]
@samdark 
Old issue but I agree with https://github.com/yiisoft/yii2/issues/15665#issuecomment-367979881 about response headers for CORS-preflight request. Response headers must contain `Access-Control-Allow-Methods`.

For example if we try to send PATCH-request from domain.com to api.domain.com (cross origin), browser send preflight request with `Access-Control-Request-Method: PATCH` and expects to see `PATCH` in response header `Access-Control-Allow-Methods`, otherwise it will block further PATCH-request:
`... has been blocked by CORS policy: Method PATCH is not allowed by Access-Control-Allow-Methods in preflight response.`

---

OptionsAction added response headers
https://github.com/yiisoft/yii2/blob/77ad6bc00847d4964a0b2a82d3b70dcd7cb5a1cf/framework/rest/OptionsAction.php#L43

but this action does not run because CorsFilter `return false` on beforeAction
https://github.com/yiisoft/yii2/blob/4f80cda7130da5259ce93fe41eb681cbb70f30be/framework/filters/Cors.php#L109-L113

**Comment by alnidok** [comment]
Sorry, it is my mistake. I don't have `Access-Control-Request-Method` in behavior config.

---

Cors call prepareHeaders before `return false`
https://github.com/yiisoft/yii2/blob/4f80cda7130da5259ce93fe41eb681cbb70f30be/framework/filters/Cors.php#L106

and add response header `Access-Control-Allow-Methods` with default config
https://github.com/yiisoft/yii2/blob/4f80cda7130da5259ce93fe41eb681cbb70f30be/framework/filters/Cors.php#L171-L173
